gen/drivers/indexes.go

```diff
@@ -0,0 +1,10 @@
+package drivers
+
+// DBIndexes lists all indexes in the database schema keyed by table name
+type DBIndexes map[string][]Index
+
+// Index represents an index in a table
+type Index struct {
+	Name    string   `yaml:"name" json:"name"`
+	Columns []string `yaml:"columns" json:"columns"`
+}
```

Comment:
We should instead reuse the `Constraint` type since they have the same structure.

Perhaps it makes more sense for the type to be called `Index` or something else, but I would rather no duplication.

Reply:
Even though both types have the same structure, an `Index` and a `Constraint` have different semantics. Indexes are automatically created for primary key columns and unique constraints, but things are different for foreign keys. MySQL (InnoDB) automatically creates indexes for them, but PostgreSQL and SQLite do not.

Take the following DDL statements, for example, executed against a PostgreSQL database:

```sql
CREATE TABLE test.parent (
   id INT GENERATED ALWAYS AS IDENTITY,
   name VARCHAR(255) NOT NULL,
   PRIMARY KEY(id)
);

CREATE TABLE test.child (
   id INT GENERATED ALWAYS AS IDENTITY,
   parent_id INT,
   name VARCHAR(255) NOT NULL,
   PRIMARY KEY(id),
   CONSTRAINT fk_parent FOREIGN KEY(parent_id) REFERENCES parent(id)
);
```

If we run the following query to list their indexes:

```sql
SELECT * FROM pg_catalog.pg_indexes WHERE schemaname = 'test';
```

We get the following result:

```text
schemaname|tablename|indexname  |tablespace|indexdef                                                       |
----------+---------+-----------+----------+---------------------------------------------------------------+
test      |parent   |parent_pkey|          |CREATE UNIQUE INDEX parent_pkey ON test.parent USING btree (id)|
test      |child    |child_pkey |          |CREATE UNIQUE INDEX child_pkey ON test.child USING btree (id)  |
```

I can certainly generalize the struct to avoid code duplication, but neither `Index` nor `Constraint` will be a semantically correct name.

Reply:
Yes, I understand that semantically, they are different things. This is why I said perhaps it should be called something different (not Index or Constraint).

It just feels a bit redundant to have two types that are exactly the same thing.

I guess a bit of repetition can be good for the sake of correctness.

Reply:
What do you think about something similar:

```go
type NamedColumnList struct {
	Name    string   `yaml:"name" json:"name"`
	Columns []string `yaml:"columns" json:"columns"`
}


type Constraint struct {
	NamedColumnList
}

type Index struct {
	NamedColumnList
}
```

Or would you rather completely eliminate the `Index` type?

Reply:
This will allow the `Index` and `Constraint` types to evolve separately (if needed).

Reply:
> This will allow the `Index` and `Constraint` types to evolve separately (if needed).

If that time comes, then we can create these wrapper types. In the meantime, we can replace every instance with `Constraint` with `NamedColumnList`.

If we really want to have distinct types now, we can do 

```go
type Constraint NamedColumnList

type Index NamedColumnList
```

This is also great since there are no methods attached to `NamedColumnList`